src/figuresClasses.ts

```diff
@@ -2,10 +1,98 @@
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+    public shape: Shape = 'triangle',
+  ) {
+    this.a = a;
+    this.b = b;
+    this.c = c;
+    this.color = color;
+    this.shape = 'triangle';
+
+    if (
+      this.a === 0 ||
+      this.b === 0 ||
+      this.c === 0 ||
+      this.a >= this.b + this.c ||
+      this.b >= this.a + this.c ||
+      this.c >= this.a + this.b
+    ) {
+      throw new Error('your error message');
+    }
+  }
+
+  getArea(): number {
+    const semiperimetrOfTriangle = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiperimetrOfTriangle *
+        (semiperimetrOfTriangle - this.a) *
+        (semiperimetrOfTriangle - this.b) *
+        (semiperimetrOfTriangle - this.c),
+    );
+
+    return Math.floor(area * 100) / 100;
+  }
+}
+
+export class Circle implements Figure {
+  constructor(
+    public color: Color,
+    public radius: number,
+    public shape: Shape = 'circle',
```

Comment:
The `shape` field is being set in the constructor with a default value, which is unnecessary because the class `Circle` should already encapsulate the idea of its shape being a circle. You can remove the default value assignment from the constructor and declare the `shape` field in the class with a default value or as a readonly property.